js/components/multi-input.js

```diff
@@ -0,0 +1,88 @@
+define([
+  'knockout',
+  'text!./multi-input.html',
+	'utils/CommonUtils',
+  'components/Component',
+  'utils/AutoBind',
+  'lodash',  
+  'databindings',
+  'less!./multi-input.less',
+], function (
+  ko, 
+    view,
+    commonUtils,
+    Component,
+    AutoBind,
+    _,
+) {
+    const typeIdentifier = {
+      integer: "integer",
+      float: "float"
+    }
+    class MultiInput extends AutoBind(Component) {
+      constructor(params) {
+        super(params);
+        this.selectedValueType = params.selectedValueType || "integer";
+        this.selectedValueExtender = params.selectedValueExtender || {};
+        this.selectedValues = params.selectedValues;
+        this.itemToAdd = ko.observable("").extend(this.selectedValueExtender);
+        this.defaultValues = params.defaultValues;
+        this.valueLabel = params.valueLabel || "Value";
+
+        this.hasSelectedValues = ko.pureComputed(() => {
+          return (this.selectedValues && this.selectedValues().length > 0);
+        });
+        this.hasDefaults = ko.pureComputed(() => {
+          return (this.defaultValues && this.defaultValues.length > 0);
+        });
+        this.enableAdd = ko.pureComputed(() => {
+         return (this.itemToAdd() !== null && this.itemToAdd().toString().length > 0 && !isNaN(this.getParser()(this.itemToAdd())));
```

Comment:
Why would you need `isNaN` if the extender would be enforced based on `
`selectedValueType`?

Reply:
So the numeric extender has to allow for some non-numeric inputs - for example, if you are typing "-1", the extender must allow you to enter "-" first before it attempts to reformat anything. Without the isNaN, it would be possible to enter "-" for a list of integers and so this check prevents that from happening.

Reply:
Makes sense. But then I would also tighten this to `selectedValueType` (as the `selectedValueExtender` which we discussed above) and abstract it into a separate method, smth like `checkValue`. So, as a suggestion, why not to introduce an interface (base class) which contains three fields: `type`, `extender`, `checkValue`, and extend that for Integer, Float, whatever. And then work via that interface.

Reply:
Sure - let me add that into this branch.